### ISSUE TITLE
Gamepad Support

### ISSUE THREAD

**Issue by Snayff** [Snayff/nqp2#178]
## Suggested Feature 
Implement gamepad support.

## Requirements 
- [x] Can navigate around options on screen using the stick or d-pad
- [x] Can interact with UI elements with gamepad button (A for xbox?)
- [x] Can cancel partially selected input with gamepad button (B for xbox?)

## Additional Context
None

**Comment by bitcraft** [comment]
working on this.  basic support is implemented.  replaces keyboard in most of the game, but combat seems to be doing its own thing.  are you aware of any limitations that would make the game unplayable without the keyboard at this time?

**Comment by bitcraft** [comment]
nvm, i did something wrong.  gamepad seems to be working for everything, i was able to use an xbox controller to play a round of combat.

**Comment by bitcraft** [comment]
controller config
```
gamepad_label_map = {
    GamepadButton.UP: "up",
    GamepadButton.LEFT: "left",
    GamepadButton.DOWN: "down",
    GamepadButton.RIGHT: "right",
    GamepadButton.EAST: "cancel",
    GamepadButton.SOUTH: "select",
    GamepadButton.NORTH: "view_troupe",
}
```
the cardinal directions are used because "a b x y" doesn't work for different brands, and ps4 controllers have shapes.  so the directions match one of the 4 buttons on the left side.  this can be customized for other gamepads, like ps3/4/5, nintendo switch, or something else (will need a gui for it).

**Comment by snayff** [comment]
@bitcraft are you intending to submit a PR for this?

**Comment by bitcraft** [comment]
yes.  i will need some time later today to clean it up some

**Comment by snayff** [comment]
closed in #186
